File: Arrays and Hashmap/two-sum.cpp
/*
* Two Sum : https://leetcode.com/problems/two-sum/description/
*/
#include <iostream>
#include <string>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> umap;
        for(int i=0;i<nums.size();i++){
            if(umap.find(target - nums[i])!=umap.end())
                return {umap[target-nums[i]],i};
            umap[nums[i]]=i;
        }
        return {};
    }
};

int main(){
    vector<int> nums;
    nums.push_back(11);
    nums.push_back(7);
    nums.push_back(10);
    nums.push_back(2);

    int target = 9;

    Solution s;

    cout<<"[";
    for(int i = 0; i<s.twoSum(nums,target).size(); i++){
        cout << s.twoSum(nums,target)[i];
        if(i < 1) 
            cout<<",";
    }
    cout<<"]";
}

File: Arrays and Hashmap/two-sum.py
'''
* Two Sum : https://leetcode.com/problems/two-sum/description/
'''
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen = {}
        for i, num in enumerate(nums):
            comp = target - num
            if comp in seen:
                return [seen[comp], i]
            seen[num] = i
        return []

# Example usage
if __name__ == "__main__":
    solution = Solution()
    nums = [2, 7, 11, 15]
    target = 9
    result = solution.twoSum(nums, target)
    print("Indices of two numbers that add up to target:", result)
